### PR TITLE
HYAS Protect Initial Commit

### PR DESCRIPTION
**Pull request by RamboV**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-rusraut** [comment]
Hi @RamboV,
Thanks for rising PR with us, we will provide you an update by 23 Aug 2023.

**Comment by v-rusraut** [comment]
Hi @RamboV,

Please work on below points:

- Please package the solution using V3 tool, please go through this readme file on how to use the V3 tool
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md

- Add ReleaseNotes file for solution,please refer below link for how to create ReleaseNotes.
https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/ReleaseNotesGuidance.md

- We need to create parser with yaml file , don't create parser with txt file,use [this](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/CrowdStrike%20Falcon%20Endpoint%20Protection/Parsers/CrowdStrikeReplicatorV2.yaml) as reference. 
- Provide  sample data for table **HYASProtectDnsSecurityLogs_CL** , we need sample data for testing parser.

Thanks

**Comment by v-rusraut** [comment]
Hi @RamboV,

Before repackaging the solution, please resolve below validation erros:

- Below hyperlink is present in HYASProtect_FunctionApp.json file, it is not working, please correct it
link : https://github.com/moin-loginsoft/HyasProtect/blob/main/Data%20Connectors/HyasProtect.zip?raw=true
HyasProtect2


- There is KqlValidation issue, please refer below screen shots.
 
HyasProtect1

To resolve this issue please add table definition of **HYASProtectDnsSecurityLogs_CL** at  below mentioned path. 
Path : https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables

Thanks

**Comment by v-rbajaj** [comment]
Hi @RamboV, can you please look into the above comment and make the necessary changes?

**Comment by v-atulyadav** [comment]
Hi @RamboV, 
We are still waiting for your response to the above. Thanks

**Comment by v-rbajaj** [comment]
Hi @RamboV, can you please provide some update on the above comments?

**Comment by RamboV** [comment]
Hi @v-rbajaj,
We are looking into it and will update you by 7 sep 2023.

**Comment by v-rbajaj** [comment]
Hi @RamboV, Ok thank, waiting for update.

**Comment by v-atulyadav** [comment]
Hi @RamboV, 
We look forward to hearing from you once your updates are complete. Thanks

**Comment by v-rbajaj** [comment]
Hi @RamboV, please provide some update on this PR.

**Comment by v-rbajaj** [comment]
Hi @RamboV, please provide update on this PR.

**Comment by v-atulyadav** [comment]
Hi @RamboV, hope you are doing well. Just wanted to check if you got a chance to look at the suggestions shared. Please feel free to reach out to us for any queries and/or support. Thanks

**Comment by v-rbajaj** [comment]
Hi @RamboV, please provide update on this PR.

**Comment by v-rbajaj** [comment]
Hi @RamboV, Please move the Logo file to https://github.com/Azure/Azure-Sentinel/tree/master/Logos

**Comment by RamboV** [comment]
> Hi @RamboV, Please move the Logo file to https://github.com/Azure/Azure-Sentinel/tree/master/Logos

Hi @v-rbajaj  we already have a logo for HYAS we are using the same.

**Comment by v-rbajaj** [comment]
Ok, can you please remove the logo from here Solutions/HYAS Protect/Data Connectors/Logo/Hyas.svg

> > Hi @RamboV, Please move the Logo file to https://github.com/Azure/Azure-Sentinel/tree/master/Logos
> 
> Hi @v-rbajaj we already have a logo for HYAS we are using the same.

**Comment by RamboV** [comment]
> Ok, can you please remove the logo from here Solutions/HYAS Protect/Data Connectors/Logo/Hyas.svg
> 
> > > Hi @RamboV, Please move the Logo file to https://github.com/Azure/Azure-Sentinel/tree/master/Logos
> > 
> > 
> > Hi @v-rbajaj we already have a logo for HYAS we are using the same.

Hello @v-rbajaj  done

File: Solutions/HYAS Protect/Data/Solution_HyasProtectLogs.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "Name": "HYAS Protect",
+    "Author": "Hyas",
+    "Logo": "<img src=\"https://raw.githubusercontent.com/Azure/Azure-Sentinel/blob/master/Logos/Hyas.svg\" width=\"75px\" height=\"75px\">",
```
Comment (v-rbajaj) [inline Solutions/HYAS Protect/Data/Solution_HyasProtectLogs.json:4]: Hi @RamboV, please update this  link https://[raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Hyas.svg](https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Logos/Hyas.svg)

File: Solutions/HYAS Protect/Package/mainTemplate.json [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+        "description": "Workspace name for Log Analytics where Microsoft Sentinel is setup"
+      }
+    }
+  },
+  "variables": {
+    "_solutionName": "HYAS Protect",
+    "_solutionVersion": "3.0.0",
+    "solutionId": "hyas.microsoft-sentinel-solution-hyas-protect",
+    "_solutionId": "[variables('solutionId')]",
+    "workspaceResourceId": "[resourceId('microsoft.OperationalInsights/Workspaces', parameters('workspace'))]",
```
Comment (v-rbajaj) [inline Solutions/HYAS Protect/Package/mainTemplate.json:37]: Hi @RamboV, please remove this line from maintemplate.json and update this file inside 3.0.0 zip file as well.

**Comment by v-rbajaj** [comment]
Hi @RamboV, please provide some update on the above comment.

**Comment by v-rbajaj** [comment]
Hi @RamboV, please provide some update on the above comment.

**Comment by RamboV** [comment]
> Hi @RamboV, please provide some update on the above comment.

Hello @v-rbajaj  I have modified accordingly

**Review by v-rbajaj — APPROVED** [review]

**Review by v-rbajaj — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**